packages/builder/src/builder.js

```diff
@@ -592,26 +597,20 @@ export default class Builder {
   }
 
   watchClient() {
-    const src = this.options.srcDir
-    const rGlob = dir => ['*', '**/*'].map(glob => r(src, `${dir}/${glob}.{${this.supportedExtensions.join(',')}}`))
-
     let patterns = [
-      r(src, this.options.dir.layouts),
-      r(src, this.options.dir.middleware),
-      ...rGlob(this.options.dir.layouts)
+      r(this.options.srcDir, this.options.dir.layouts),
+      r(this.options.srcDir, this.options.dir.middleware)
     ]
+
     if (this.options.store) {
-      patterns.push(r(src, this.options.dir.store))
+      patterns.push(r(this.options.srcDir, this.options.dir.store))
     }
 
     if (this._nuxtPages && !this._defaultPage) {
-      patterns.push(
-        r(src, this.options.dir.pages),
-        ...rGlob(this.options.dir.pages)
-      )
+      patterns.push(r(this.options.srcDir, this.options.dir.pages))
     }
 
-    patterns = patterns.map(upath.normalizeSafe)
+    patterns = patterns.map((path, ...args) => upath.normalizeSafe(this.globPathWithExtensions(path), ...args))
```

Comment:
As we are mapping to glob does it still covers directory creation? (`store`, `pages`)

Reply:
Yes, if they dont exists yet they are added in `watchRestart`: https://github.com/nuxt/nuxt.js/blob/4cefc7bb893c74829a466c9f0ea9edcf61afec8e/packages/builder/src/builder.js#L660-L666